src/mpi/coll/allreduce/allreduce.c

```diff
@@ -207,13 +207,19 @@ int MPIR_Allreduce(const void *sendbuf, void *recvbuf, MPI_Aint count, MPI_Datat
     void *in_recvbuf = recvbuf;
     void *host_sendbuf;
     void *host_recvbuf;
+    int is_supported = 0;
 
-    MPIR_Coll_host_buffer_alloc(sendbuf, recvbuf, count, datatype, &host_sendbuf, &host_recvbuf);
-    if (host_sendbuf)
-        sendbuf = host_sendbuf;
-    if (host_recvbuf)
-        recvbuf = host_recvbuf;
-
+    if (MPIR_CVAR_ENABLE_YAKSA_REDUCTION) {
+        is_supported = MPIR_Typerep_reduce_is_supported(op, datatype);
+    }
```

Comment:
`is_supported` is a confusing name here. The purpose is to decide whether need host buffer, so why not name it `need_host_buffer`?